test/parsers/string.test.ts

```diff
@@ -491,8 +491,67 @@ suite("String validations", (test) => {
       ),
       {
         type: "string",
-        pattern: zodPatterns.email,
+        pattern: zodPatterns.email.source,
       },
     );
   });
+
+  test("should correctly handle reasonable non-contrived regexes with flags", (assert) => {
+    assert(
+      parseStringDef(
+        z.string().regex(/(^|\^foo)Ba[r-z]+./)._def,
+        getRefs({ applyRegexFlags: true })
+      ),
+      {
+        type: "string",
+        pattern: "(^|\\^foo)Ba[r-z]+.",
+      }
+    );
+
+    assert(
+      parseStringDef(
+        z.string().regex(/(^|\^foo)Ba[r-z]+./i)._def,
+        getRefs({ applyRegexFlags: true })
+      ),
+      {
+        type: "string",
+        pattern: "(^|\\^[fF][oO][oO])[bB][aA][r-zR-Z]+.",
+      }
+    );
+
+    assert(
+      parseStringDef(
+        z.string().regex(/(^|\^foo)Ba[r-z]+./ms)._def,
+        getRefs({ applyRegexFlags: true })
+      ),
+      {
+        type: "string",
+        pattern: "((^|(?<=[\r\n]))|\\^foo)Ba[r-z]+[.\r\n]",
+      }
+    );
+
+    assert(
+      parseStringDef(
+        z.string().regex(/(^|\^foo)Ba[r-z]+./ims)._def,
+        getRefs({ applyRegexFlags: true })
+      ),
+      {
+        type: "string",
+        pattern: "((^|(?<=[\r\n]))|\\^[fF][oO][oO])[bB][aA][r-zR-Z]+[.\r\n]",
+      }
+    );
+  });
+
+  test("should fall back to regex source if a conversion can't be made", (assert) => {
+    assert(
+      parseStringDef(
+        z.string().regex(/[Z-a]/i)._def,
+        getRefs({ applyRegexFlags: true })
+      ),
+      {
+        type: "string",
+        pattern: "[Z-a]",
+      }
+    );
+  });
```

Comment:
Please note that this test will a warning whenever it's run. This is intended behaviour (since we'd like the user to know if their package isn't doing what they expect!), but it does make the test suite a little more ugly on the dev side...

If there's a cleaner way to do this (e.g. make `suite()` run tests silently), please let me know!

Reply:
I can drop it before release.